tests/Unit/CallPowerTest.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace Tests\Unit;
+
+use Exception;
+use Tests\TestCase;
+use Chompy\Services\Rogue;
+
+class CallPowerTest extends TestCase
+{
+    /**
+     * Test that a post with a completed status successfully sends to Rogue.
+     *
+     * @return void
+     */
+    public function testCompletedCallStatus()
+    {
+        // Create a post with completed as the status.
+        $response = $this->withChompyApiKey()->json('POST', 'api/v1/callpower/call', [
+            'mobile' => '1234567891',
+            'callpower_campaign_id' => '1',
+            'status' => 'completed',
+            'call_timestamp' => '2017-11-09 06:34:01.185035',
+            'call_duration' => 50,
+            'campaign_target_name' => 'Mickey Mouse',
+            'campaign_target_title' => 'Representative',
+            'campaign_target_district' => 'FL-7',
+            'callpower_campaign_name' => 'Test',
+            'number_dialed_into' => '+12028519273',
+        ]);
+
+        $response->assertStatus(200);
+    }
+
+    /**
+     * Test that a post with a busy status successfully sends to Rogue.
+     *
+     * @return void
+     */
+    public function testBusyCallStatus()
+    {
+        // Create a post with completed as the status.
+        $response = $this->withChompyApiKey()->json('POST', 'api/v1/callpower/call', [
+            'mobile' => '1234567891',
+            'callpower_campaign_id' => '1',
+            'status' => 'busy',
+            'call_timestamp' => '2017-11-09 06:34:01.185035',
+            'call_duration' => 50,
+            'campaign_target_name' => 'Mickey Mouse',
+            'campaign_target_title' => 'Representative',
+            'campaign_target_district' => 'FL-7',
+            'callpower_campaign_name' => 'Test',
+            'number_dialed_into' => '+12028519273',
+        ]);
+
+        $response->assertStatus(200);
+    }
+
+    /**
+     * Test that a post with an invalid CallPower campaign id is not made,
+     * and enters into the failed_jobs table.
+     *
+     * @return void
+     */
+    public function testFailedPostWithInvalidCallPowerCampaignId()
+    {
+        // Mock the Rogue API call to get the action id.
+        $this->mock(Rogue::class)
```

Comment:
I think this totally is on the right track! These tests can probably be whittled down to the job itself (since we test the validation and authorization components above). So, maybe something like:

```js
// If the provided Action ID doesn't exist in Rogue, this method throws:
$this->rogueMock->shouldReceive('getActionIdFromCallPowerCampaignId')->andThrow(new Exception(500));

// Since we don't have an action in Rogue, we shouldn't make a post:
$this->rogueMock->shouldNotReceive('createPost');

CreateCallPowerPostInRogue::dispatch([
  // ...
]);
```